odbc/src/api/data.rs

```diff
@@ -1299,9 +1295,9 @@ pub mod isize_len {
 /// # Safety
 /// This writes to a raw C-pointers
 ///
-pub unsafe fn set_str_length(string_length_ptr: *mut Integer, length: Integer) {
-    if !string_length_ptr.is_null() {
-        *string_length_ptr = length
+pub unsafe fn set_data_length<T>(data_length_ptr: *mut T, length: T) {
```

Comment:
I figured I should rename this because it's now used more generally.

Reply:
yes please.
Something like `ptr_write_safe` maybe?

Reply:
Ok. I'll go with `len_ptr_safe_write` as suggested by Nathan.

Reply:
Could we please not have the `len` prefix in the helper name?

That you are writing length info to the pointer in this PR does not reflect what this is for. We want to make sure that a pointer is not Null before we write data into where it points to or the driver will blow up. 
I did not ask you to protect the other location like in set_output_binary_helper where we write because we are already checking that the pointer is not null just above. I am wondering if we should anyway though, so that we get into the best practice of always using the safe method and won't have any surprises. This ties to Patrick's about how to know if we missed a location. I don't have a good answer for that, it's a best effort deal. Making sure that in the future we always use the safe method, will help protecting ourselves and avoid mistakes later on.


In short, the idea behind this change is to have a helper which we should use **every time we write data to a pointer** no matter what the data is.

Reply:
Ok. That makes sense. I'll change it.

Reply:
Actually I'm a little confused by your comment. How would we use `ptr_safe_write` in `set_output_binary_helper`?
I'm not sure how it would fit in.

Reply:
umh, let me check again. Ah, yes, this is a buffer write. The same idea actually applies, but I see why don't see how it fits in.
In functions.rs, there are better examples. Sorry about that.
In sql_get_stmt_attrw_helper for example. The Null check is actually done earlier in SQLGetStmtAttrW. We assume (correctly in this case) that value_ptr never null.
```
StatementAttribute::SQL_ATTR_CURSOR_SCROLLABLE => {
                *(value_ptr as *mut CursorScrollable) =
                    stmt.attributes.read().unwrap().cursor_scrollable;
                SqlReturn::SUCCESS
            }
```

Reply:
Ok I see what you mean. So is your idea that we replace every instance of setting a pointer (in certain files like `data.rs`, `functions.rs`, etc) with `pointer_safe_write` regardless` of whether it was checked earlier? Essentially to mitigate any mistakes.

Reply:
Exactly 😀